### PR TITLE
RLP encoding improvements

### PR DESCRIPTION
**Pull request by gtrepta**
This is a WIP for some improvements to the semantics involving rlp. The primary goal is to create some utility functions that can compress semantics rules into fewer lines, improve performance on existing rlp rules, and potentially remove some redundant rlp encoding rules.

- [x] Create an `#rlpEncode( JSON )` function which can encode a JSON list (encoding of lists is a common pattern)
- [x] Use the new rlpEncode function in rules which can benefit from it
- [x] Use `StringBuffer` sort instead of `String` to build rlp strings (`rlpEncode{Logs,Topics,JSONBytes,AccessList}`)
- [x] Find and remove redundant rlp encoding rules (`rlpEncode{Transaction,JSONBytes}`??)

### CODE REVIEW

**Comment by ehildenb** [comment]
Wow this looks really great so far :)

**Comment by gtrepta** [comment]
I'm marking this ready for review now. There are still some rlp encoding functions that can't take advantage of the new utilities because the data they encode is structured a bit too heavily. These are `#rlpEncodeLogs` and `#rlpEncodeMerkleTree`.

File: serialization.md [hunk 1]
```diff
@@ -285,6 +237,18 @@ We need to interperet a `ByteArray` as a `String` again so that we can call `Kec
...
+- `#addrBytes` Takes an Account and represents it as a 20-byte wide ByteArray (or an empty ByteArray for a null address)
+- `#wordBytes` Takes an Int and represents it as a 32-byte wide ByteArray
+
+```k
+    syntax ByteArray ::= #addrBytes( Account ) [function]
+                       | #wordBytes( Int )     [function]
+ // -----------------------------------------------------
+    rule #addrBytes(.Account) => .ByteArray
+    rule #addrBytes(ACCT)     => #padToWidth(20, #asByteStack(ACCT))
+    rule #wordBytes(WORD)     => #padToWidth(32, #asByteStack(WORD))
```
Comment (ehildenb) [inline serialization.md:249]: Should these have validity constraints on the inputs? Like `#rangeAddress(ACCT)` and `#rangeUInt(256, WORD)`?

**Review by ehildenb — APPROVED** [review]

File: serialization.md (hunk 1 above)
Comment (gtrepta) [reply serialization.md]: Sounds like a good idea. It could help catch some bugs in the future.

**Pull request merged**